pkg/server/endpoints/harvester.go

```diff
@@ -4,98 +4,187 @@ import (
+func (h *HarvesterAPIHandlers) GetRelationships(ctx echo.Context, params harvester.GetRelationshipsParams) error {
 	return nil
 }
 
 // PatchRelationshipsRelationshipID accept/denied relationships requests - (PATCH /relationships/{relationshipID})
-func (h HarvesterAPIHandlers) PatchRelationshipsRelationshipID(ctx echo.Context, relationshipID api.UUID) error {
+func (h *HarvesterAPIHandlers) PatchRelationshipsRelationshipID(ctx echo.Context, relationshipID api.UUID) error {
 	return nil
 }
 
-// Onboard authenticate a trust domain in galadriel server providing its access token - (POST /trust-domain/onboard)
-func (h HarvesterAPIHandlers) Onboard(ctx echo.Context) error {
-	return nil
-}
+// Onboard introduces a harvester to Galadriel Server providing its join token, and gets back a JWT token - (GET /trust-domain/onboard)
+func (h *HarvesterAPIHandlers) Onboard(echoCtx echo.Context, params harvester.OnboardParams) error {
+	ctx := echoCtx.Request().Context()
 
-// BundleSync synchronize the status of trust bundles between server and harvester - (POST /trust-domain/{trustDomainName}/bundles/sync)
-func (h HarvesterAPIHandlers) BundleSync(ctx echo.Context, trustDomainName api.TrustDomainName) error {
-	return nil
+	token, err := h.Datastore.FindJoinToken(ctx, params.JoinToken)
+	if err != nil {
+		msg := errors.New("error looking up token")
+		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
+	}
+
+	if token == nil {
+		err := errors.New("token not found")
+		return h.handleErrorAndLog(err, err, http.StatusBadRequest)
+	}
+
+	if token.ExpiresAt.Before(time.Now()) {
+		err := fmt.Errorf("token expired: trust domain ID: %s", token.TrustDomainID)
+		msg := fmt.Errorf("token expired")
+		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
+	}
+
+	if token.Used {
+		err := fmt.Errorf("token already used: trust domain ID: %s", token.TrustDomainID)
+		msg := fmt.Errorf("token already used")
+		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
+	}
+
+	trustDomain, err := h.Datastore.FindTrustDomainByID(ctx, token.TrustDomainID)
+	if err != nil {
+		msg := fmt.Errorf("error looking up trust domain")
+		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
+	}
+
+	// mark token as used
+	_, err = h.Datastore.UpdateJoinToken(ctx, token.ID.UUID, true)
+	if err != nil {
+		msg := fmt.Errorf("internal error")
+		return h.handleErrorAndLog(err, msg, http.StatusInternalServerError)
+	}
+
+	h.Logger.Infof("Received onboard request from trust domain: %s", trustDomain.Name)
+
+	jwtParams := &jwt.JWTParams{
+		Issuer:   constants.GaladrielServerName,
+		Subject:  trustDomain.Name,
+		Audience: []string{constants.GaladrielServerName},
+		TTL:      24 * 5 * time.Hour,
+	}
+
+	jwtToken, err := h.jwtIssuer.IssueJWT(ctx, jwtParams)
+	if err != nil {
+		msg := fmt.Errorf("error generating JWT token")
+		return h.handleErrorAndLog(err, msg, http.StatusInternalServerError)
+	}
+
+	return echoCtx.JSON(http.StatusOK, jwtToken)
```

Comment:
You should use the wrapped methods in our http package. Something like: 
```suggestion
	return chttp.WriteResponse(echoCtx, jwtToken)
```